### DISCUSSION TITLE
Prisma Client Generate

### DISCUSSION THREAD

**Discussion by rorn-prometeus** [prisma/prisma#21624] [Q&A]
### Question
```
FROM node:alpine As development

WORKDIR /usr/src/app

COPY package*.json ./
COPY libs ./
COPY prisma ./

RUN npm install
COPY . .

RUN npx prisma generate
RUN npm run build gateway

FROM node:alpine as production

ARG NODE_ENV=production
ENV NODE_ENV=${NODE_ENV}

WORKDIR /usr/src/app

COPY package*.json ./

RUN npm install --omit=dev

COPY . .

COPY --from=development /usr/src/app/dist ./dist
RUN ls -l dist

CMD ["node", "dist/apps/gateway/main"]
```
When i run docker build point to production it say prisma client not generate but in development it work fine

### Expected behavior (optional)

please correct me if i am wrong

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @rorn-prometeus 👋 

Thank you for raising this question.

In your Dockerfile, you're running `npx prisma generate` in the development stage, but the generated Prisma Client is not being copied over to the production stage. This is because the Prisma Client is generated in the `node_modules/.prisma/client` directory, which is not being copied over to the production stage.

To resolve this issue, you can add a `postinstall` script in your `package.json` file to ensure Prisma Client is generated during the installation of node modules in the production stage. Here's how you can do it:
```
{ ... "scripts" { "postinstall": "prisma generate" } ...}
```
This script will run prisma generate every time npm install is run, ensuring that Prisma Client is always up-to-date.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by ludralph** [reply]
Hi @rorn-prometeus 

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
